RudderAnalytics/Request/BlockingRequestHandler.cs

```diff
@@ -222,8 +229,13 @@ public async Task MakeRequest(Batch batch)
                         watch.Stop();
 
                         var response = (HttpWebResponse)ex.Response;
+                        if(response != null && response.GetResponseStream() != null) {
+                        using (var reader = new System.IO.StreamReader(response.GetResponseStream(), ASCIIEncoding.ASCII)) {
+                            responseStr = reader.ReadToEnd(); 
+                        }
+                        }
                         statusCode = (response != null) ? (int)response.StatusCode : 0;
-                        if ((statusCode >= 500 && statusCode <= 600) || statusCode == 429 || statusCode == 0)
+                      if ((statusCode >= 500 && statusCode <= 600) || statusCode == 429 || statusCode == 0)
                         {
                             // If status code is greater than 500 and less than 600, it indicates server error
                             // Error code 429 indicates rate limited.
```

Comment:
can we check with the server if we indeed sent 429 for rate limited responses?

Reply:
yes, server does send that confirmed with the team